dotCMS/src/main/java/com/dotcms/rest/IntegrityResource.java

```diff
@@ -153,68 +149,99 @@ private Response postWithEndpointState(String endpointId, String url, MediaType
-
-
-
-            remoteIP = request.getRemoteHost();
-            if(!UtilMethods.isSet(remoteIP))
-                remoteIP = request.getRemoteAddr();
-
-            PublishingEndPointAPI endpointAPI = APILocator.getPublisherEndPointAPI();
-            final PublishingEndPoint requesterEndPoint = endpointAPI.findEnabledSendingEndPointByAddress(remoteIP);
-
-            if(!BundlePublisherResource.isValidToken(auth_token_digest, remoteIP, requesterEndPoint)) {
+            final PublishingEndPoint requesterEndpoint = endpointAPI.findEnabledSendingEndPointByAddress(remoteIp);
+            if (!BundlePublisherResource.isValidToken(auth_token_digest, remoteIp, requesterEndpoint)) {
+                Logger.error(
+                        IntegrityResource.class,
+                        String.format(
+                                "Receiver at %s:> Authentication Token is invalid for ip: %s and endpoint id %s",
+                                localAddress,
+                                remoteIp,
+                                requesterEndpoint.getId()));
                 return Response.status(HttpStatus.SC_UNAUTHORIZED).build();
             }
 
-            ServletContext servletContext = request.getSession().getServletContext();
-
-            if(servletContext.getAttribute("integrityRunning")!=null && ((Boolean) servletContext.getAttribute("integrityRunning"))) {
-                throw new WebApplicationException(Response.status(HttpStatus.SC_CONFLICT).entity("Already Running").build());
+            if (QuartzUtils.isJobRunning(
+                    IntegrityDataGenerationJob.JOB_NAME,
+                    IntegrityDataGenerationJob.JOB_GROUP)) {
+                Logger.error(
+                        IntegrityResource.class,
+                        String.format(
+                                "Receiver at %s:> job is already running for endpoint id: %s, so aborting generation",
+                                localAddress,
+                                requesterEndpoint.getId()));
+                throw new WebApplicationException(
+                        Response.status(HttpStatus.SC_CONFLICT)
+                                .entity("Already Running")
+                                .build());
             }
 
-            String transactionId = UUIDGenerator.generateUuid();
-            servletContext.setAttribute("integrityDataRequestID", transactionId);
-
-            // start data generation process
-            IntegrityDataGeneratorThread idg = new IntegrityDataGeneratorThread( requesterEndPoint, request.getSession().getServletContext() );
-            idg.start();
-            //Saving the thread on the session context for a later use
-            servletContext.setAttribute( "integrityDataGeneratorThread_" + transactionId, idg );
+            final String transactionId = UUIDGenerator.generateUuid();
+            IntegrityDataGenerationJob.triggerIntegrityDataGeneration(requesterEndpoint, transactionId);
```

Comment:
I think that the triggerIntegrityDataGeneration method should check if the Quartz is running inner, I mean this code:

https://github.com/dotCMS/core/pull/19243/files#diff-1c1ba904ed03287c1c0b48a6f3971449R204

and if the quartz is running throw a exception then in the resource catch the exception and send the response

Reply:
Isn't Quartz in charge of checking that for me?